ta4j/src/main/java/org/ta4j/core/indicators/pivotpoints/FibonacciReversalIndicator.java

```diff
@@ -0,0 +1,96 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.pivotpoints;
+
+import org.ta4j.core.Decimal;
+import org.ta4j.core.indicators.RecursiveCachedIndicator;
+
+import java.util.List;
+
+/**
+ * Fibonacci Reversal Indicator.
+ * <p>
+ * @author team172011(Simon-Justus Wimmer), 09.10.2017
+ * @see <a href="http://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:pivot_points">chart_school: pivotpoints</a>
+ */
+public class FibonacciReversalIndicator extends RecursiveCachedIndicator<Decimal> {
+
+    private PivotPointIndicator pivotPointIndicator;
+    private PivotLevel level;
+
+    public enum PivotLevel{
+        RESISTANCE_3(Decimal.ONE, true),
+        RESISTANCE_2(Decimal.valueOf(0.618), true),
+        RESISTANCE_1(Decimal.valueOf(0.382), true),
+        SUPPORT_1(Decimal.valueOf(0.382), false),
+        SUPPORT_2(Decimal.valueOf(0.618), false),
+        SUPPORT_3(Decimal.ONE, false);
+
+        private Decimal fibFactor;
+        private boolean isResistance;
+
+        PivotLevel(Decimal fibFactor, boolean isResistance){
+            this.fibFactor = fibFactor;
+            this.isResistance = isResistance;
+        }
+
+        public Decimal getFactor(){
+            return this.fibFactor;
+        }
+
+        public boolean isResistance(){
+            return isResistance;
+        }
+
+    }
+
+
+    /**Constructor.
+     * <p>
+     * Calculates the (fibonacci) support 3 reversal
+     * @param pivotPointIndicator the {@link PivotPointIndicator} for this reversal
+     */
+    public FibonacciReversalIndicator(PivotPointIndicator pivotPointIndicator, PivotLevel level) {
+        super(pivotPointIndicator);
+        this.pivotPointIndicator = pivotPointIndicator;
+        this.level = level;
+    }
+
+    @Override
+    protected Decimal calculate(int index) {
+        List<Integer> ticksOfPreviousPeriod = pivotPointIndicator.getTicksOfPreviousPeriod(index);
+        if (ticksOfPreviousPeriod.isEmpty())
+            return Decimal.NaN;
+        Decimal high =  getTimeSeries().getTick(ticksOfPreviousPeriod.get(0)).getMaxPrice();
+        Decimal low = getTimeSeries().getTick(ticksOfPreviousPeriod.get(0)).getMinPrice();
+        for(int i: ticksOfPreviousPeriod){
+            high = (getTimeSeries().getTick(i).getMaxPrice()).max(high);
+            low = (getTimeSeries().getTick(i).getMinPrice()).min(low);
+        }
+
+        if (level.isResistance())
+            return pivotPointIndicator.getValue(index).plus(level.getFactor().multipliedBy(high.minus(low)));
```

Comment:
dont use short if. Use always curly braces. It is better maintainable. Imagine putting anything within if afterwards, then you have to put curly braces. 

Use 

if (level.isResistance()) {
   return pivotPoint.getValue(index).plus(level.getFactor().multipliedBy(high.minus(low)));
}